### PR TITLE
[fpga] add option to stub out otbn

### PR DESCRIPTION
**Pull request by tjaychen**
Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
This was a suggestion courtesy of @dsandrs . Originally I had different logic within the `stub` case vs not.
However, that changed the hierarchy of various signals in `otbn` and led to dv failing.

I'm trying this approach to see if it vivado will optimize it out. If it does, it's less disruptive to the module.
I think the ideal solution would be to do this inside `rstmgr` so the modules are completely undisturbed.

File: test/systemtest/config.py [hunk 1]
```diff
@@ -34,14 +35,14 @@
...
-        "targets": ["fpga_nexysvideo"],
-    },
+    #{
+    #    "name": "otbn_rsa_test",
+    #    "targets": ["fpga_nexysvideo"],
+    #},
+    #{
+    #    "name": "otbn_ecdsa_p256_test",
+    #    "targets": ["fpga_nexysvideo"],
+    #},
```
Comment (vogelpi) [inline test/systemtest/config.py:45]: I don't know how long these tests actually take in simulation. Since these are top-level end-to-end tests I think we should keep running them somewhere. Would it be feasible to run them in simulation until we enable OTBN on FPGA again? @imphil @rswarbrick @GregAC

**Review by vogelpi — APPROVED** [review]
Thanks @tjaychen for preparing this PR. The changes are much less invasive than I thought. This looks good to me for unblocking the silver work.

I am wondering if it would be feasible to keep the end-to-end tests running in simulation until we can enable OTBN again on FPGA. But maybe their runtime in Verilator is just too long.

If needed, I am also okay to extend this to AES but I think the impact there will be much lower as we anyway disable masking on the NexysVideo.

**Comment by imphil** [comment]
> I am wondering if it would be feasible to keep the end-to-end tests running in simulation until we can enable OTBN again on FPGA. But maybe their runtime in Verilator is just too long.

No, the runtime is significantly too long.

**Comment by vogelpi** [comment]
> > I am wondering if it would be feasible to keep the end-to-end tests running in simulation until we can enable OTBN again on FPGA. But maybe their runtime in Verilator is just too long.
> 
> No, the runtime is significantly too long.

File: hw/ip/otbn/rtl/otbn.sv [hunk 2]
```diff
@@ -233,12 +243,12 @@ module otbn
...
-  // error.
+  // adapter expects the original ECC format. Send imem_rerror as bit 1, signalling an
+  // uncorrectable error.
   //
-  // The mux ensures that imem_rerror doesn't appear on the bus (possibly leaking information) when
-  // the core is operating. Since rerror depends on rvalid, we could avoid this mux. However that
-  // seems a bit fragile, so we err on the side of caution.
+  // The mux ensures that imem_rerror doesn't appear on the bus (possibly leaking information)
+  // when the core is operating. Since rerror depends on rvalid, we could avoid this mux. However
+  // that seems a bit fragile, so we err on the side of caution.
```
Comment (imphil) [inline hw/ip/otbn/rtl/otbn.sv:251]: These changes are unrelated and should be removed.

**Review by imphil — APPROVED** [review]
LGTM if it works :-)

**Comment by imphil** [comment]
Maybe @vogelpi you can take over this PR, remove the unrelated change and this way merge it before the morning in the US?

**Comment by vogelpi** [comment]
Yes, I can do that @imphil .

File: hw/ip/otbn/rtl/otbn.sv (hunk 2 above)
Comment (vogelpi) [reply hw/ip/otbn/rtl/otbn.sv]: The first one is actually fixing a long line (>100 chars). I've now factored that into a separate commit.

Comment (imphil) [reply hw/ip/otbn/rtl/otbn.sv]: :+1: If we want to revert this commit it needs to be as focused as possible.

**Comment by msfschaffner** [comment]
Thanks for fast tracking this, guys! That really helps us.

File: hw/ip/otbn/rtl/otbn.sv (hunk 2 above)
Comment (tjaychen) [reply hw/ip/otbn/rtl/otbn.sv]: yeah let me take it out.  Originally I put a huge if statement around all the Otbn logic.  So this caused everything to increase by "2". So then I did the line fixes.

Now that we don't need it anymore, I will revert the line changes.

**Comment by tjaychen** [comment]
ah thanks for pushing this through! sorry just noticed it was merged.
